### PR TITLE
Format to current rustfmt standards and general type cleanup

### PR DESCRIPTION
**Pull request by Razaekel**
Just the results of running rustfmt on the codebase. This will set everyting to a consistent standard for future development.

### CODE REVIEW

File: src/cell.rs [hunk 1]
```diff
@@ -75,97 +84,101 @@ pub fn range_manhattan4<T: Float>(p1: math::Point4<T>, p2: math::Point4<T>) -> T
...
+        8 => [zero, diag, diag],
+        9 => [zero, diag, -diag],
+        10 => [zero, -diag, diag],
+        11 => [zero, -diag, -diag],
+        12 => [one, zero, zero],
+        13 => [zero, one, zero],
+        14 => [zero, zero, one],
+        15 => [-one, zero, zero],
+        16 => [zero, -one, zero],
+        17 => [zero, zero, -one],
```
Comment (bjz) [inline src/cell.rs]: Damn, this makes things harder to read now :(

Comment (bjz) [inline src/cell.rs]: Do you know of a way of disabling it for more 'grid-like' bits of code? An attribute?

**Comment by bjz** [comment]
Methinks I would like these in the `rustfmt.toml`:

```toml
where_trailing_comma = true
reorder_imports = true
match_block_trailing_comma = true
```

**Comment by bjz** [comment]
I wonder if we could also set up travis to fail the build if formatting changes are detected?

**Comment by Razaekel** [comment]
Went through the formatter again, with your requested config options. I had to do some manual formatting before and after running rustfmt to keep things clean and readable, mostly in cell.rs and gradient.rs.

I also added skips to the manually formatted stuff, so that rustfmt won't try to format them again and break their current layout.

rustfmt will complain about lines in the skipped sections being too long, but it will format the rest fine.

**Comment by Razaekel** [comment]
Rebased to current master, redid the format, and then went through and did a general type usage cleanup.

File: src/cell.rs [hunk 2]
```diff
@@ -74,11 +83,12 @@ pub fn range_manhattan4<T: Float>(p1: math::Point4<T>, p2: math::Point4<T>) -> T
 }
 
 #[inline(always)]
+#[cfg_attr(rustfmt, rustfmt_skip)]
```
Comment (bjz) [inline src/cell.rs]: :+1:

File: src/math.rs [hunk 3]
```diff
@@ -59,17 +61,26 @@ pub fn map4<T: Copy, U, F: Fn(T) -> U>(a: Vector4<T>, f: F) -> Vector4<U> {
...
+pub fn zip_with2<T: Copy, U: Copy, V, F: Fn(T, U) -> V>(a: Vector2<T>,
+                                                        b: Vector2<U>,
+                                                        f: F)
+                                                        -> Vector2<V> {
     let (ax, ay) = (a[0], a[1]);
     let (bx, by) = (b[0], b[1]);
     [f(ax, bx), f(ay, by)]
 }
-pub fn zip_with3<T: Copy, U: Copy, V, F: Fn(T, U) -> V>(a: Vector3<T>, b: Vector3<U>, f: F) -> Vector3<V> {
+pub fn zip_with3<T: Copy, U: Copy, V, F: Fn(T, U) -> V>(a: Vector3<T>,
```
Comment (bjz) [inline src/math.rs]: Maybe using `where` for these bounds would clean this up?

File: src/cell.rs [hunk 4]
```diff
@@ -281,17 +316,23 @@ fn cell4_seed<T, F>(seed: &Seed, point: &math::Point4<T>, range_func: F) -> (mat
...
-    let near = [whole[0] + (x_half as i64), whole[1] + (y_half as i64), whole[2] + (z_half as i64), whole[3] + (w_half as i64)];
-    let far = [whole[0] + (!x_half as i64), whole[1] + (!y_half as i64), whole[2] + (!z_half as i64), whole[3] + (!w_half as i64)];
+    let near = [whole[0] + (x_half as i64),
+                whole[1] + (y_half as i64),
+                whole[2] + (z_half as i64),
+                whole[3] + (w_half as i64)];
+    let far = [whole[0] + (!x_half as i64),
+               whole[1] + (!y_half as i64),
+               whole[2] + (!z_half as i64),
+               whole[3] + (!w_half as i64)];
```
Comment (bjz) [inline src/cell.rs]: Is there a rustfmt attribute that allows these to be formatted like:

```rust
let far = [
    whole[0] + (!x_half as i64),
    whole[1] + (!y_half as i64),
    whole[2] + (!z_half as i64),
    whole[3] + (!w_half as i64),
];

Comment (Razaekel) [inline src/cell.rs]: Unfortunately not at this time.

**Comment by bjz** [comment]
Whilst I'm not happy with everything rustfmt does, I reckon we should just merge once this is rebased

**Pull request merged**